### PR TITLE
Move peers management to memory instead of database

### PR DESCRIPTION
**Pull request by 4miners**

### CODE REVIEW

File: modules/peers.js [hunk 1]
```diff
@@ -122,17 +240,16 @@ __private.updatePeersList = function (cb) {
 					library.logger.error(['Rejecting peer', peer.string, 'with incompatible version', peer.version].join(' '));
 					self.remove(peer.ip, peer.port);
 				} else {
-					delete peer.broadhash;
-					delete peer.height;
-					self.update(peer);
+					// We currently don't base on those data, but only discover new peers here - so insert-only
```
Comment (karmacoma) [inline modules/peers.js]: `We currently don't base on those data, but only discover new peers here - so insert-only` > `We only discover new peers here - so insert-only`

Not sure what you mean by: `We currently don't base on those data`.

File: modules/peers.js [hunk 2]
```diff
@@ -79,38 +221,14 @@ __private.updatePeersList = function (cb) {
...
-		// To fine tune: decreasing random value threshold -> reduce noise.
-		if (Math.random() < 0.5) { // Every 60/0.5 = 120s
-			// Remove the first element,
-			// i.e. the one that have been placed first.
-			removed.shift();
-			removed.pop();
-		}
-
-		library.logger.debug(['Picked', picked.length, 'of', peers.length, 'peers'].join(' '));
+		library.logger.debug('Picked ' + picked.length + ' of ' + peers.length + ' peers');
```
Comment (karmacoma) [inline modules/peers.js]: Can we stick to joining array for now? This standard set throughout. We can change all separately if deemed better.

File: modules/peers.js [hunk 3]
```diff
@@ -1,35 +1,32 @@
...
-var async = require('async');
 var constants = require('../helpers/constants.js');
 var extend = require('extend');
 var fs = require('fs');
 var ip = require('ip');
 var OrderBy = require('../helpers/orderBy.js');
 var path = require('path');
 var Peer = require('../logic/peer.js');
-var PeerSweeper = require('../logic/peerSweeper.js');
+var pgp = require('pg-promise')(); // We also initialize that library here
```
Comment (karmacoma) [inline modules/peers.js]: `We also initialize that library here` > `We also initialize library here`

We need to initialize?

File: modules/peers.js [hunk 4]
```diff
@@ -292,7 +371,7 @@ Peers.prototype.acceptable = function (peers) {
 	return _.chain(peers).filter(function (peer) {
 		// Removing peers with private or host's ip address
 		return !(ip.isPrivate(peer.ip) || ip.address('public', 'ipv4', true).some(function (address) {
-			return [address, library.config.port].join(':') === [peer.ip, peer.port].join(':');
+			return (address + ':' + library.config.port) === (peer.ip + ':' + peer.port);
```
Comment (karmacoma) [inline modules/peers.js]: Can we stick to joining array for now? This standard set throughout. We can change all separately if deemed better.

File: modules/peers.js [hunk 5]
```diff
@@ -304,139 +383,228 @@ Peers.prototype.acceptable = function (peers) {
...
 				return setImmediate(waterCb, null, peers);
 			}
 		}
 	], function (err, peers) {
+		// Calculate consensus
 		var consensus = Math.round(options.matched / peers.length * 100 * 1e2) / 1e2;
 		    consensus = isNaN(consensus) ? 0 : consensus;
 
-		library.logger.debug(['Listing', peers.length, 'total peers'].join(' '));
+		library.logger.debug('Listing ' + peers.length + ' total peers');
```
Comment (karmacoma) [inline modules/peers.js]: Can we stick to joining array for now? This standard set throughout. We can change all separately if deemed better.

File: modules/peers.js [hunk 6]
```diff
@@ -304,139 +383,228 @@ Peers.prototype.acceptable = function (peers) {
...
+	// Generating insert query
+	var insert_peers = pgp.helpers.insert(peers, cs);
+
+	// Wrap sql queries in transaction and execute
+	library.db.tx(function (t) {
+		var queries = [
+        	// Clear peers table
+            t.none(sql.clear),
+            // Insert all peers
+            t.none(insert_peers)
```
Comment (karmacoma) [inline modules/peers.js]: Please check indent here.

File: modules/peers.js [hunk 7]
```diff
@@ -304,139 +383,228 @@ Peers.prototype.acceptable = function (peers) {
...
 		insertSeeds: function (seriesCb) {
-			async.eachSeries(library.config.peers.list, function (peer, eachCb) {
+			steed.each(library.config.peers.list, function (peer, eachCb) {
 				self.update({
 					ip: peer.ip,
 					port: peer.port,
 					version: modules.system.getVersion(),
 					state: 2,
-					broadhash: modules.system.getBroadhash(),
+					// Apply our broadhash to seeds cause issues and sometimes will prevent sync
```
Comment (karmacoma) [inline modules/peers.js]: `Apply our broadhash to seeds cause issues and sometimes will prevent sync` > `Apply our broadhash to seeds as can cause issues and sometimes will prevent sync`.

File: modules/peers.js [hunk 8]
```diff
@@ -304,139 +383,228 @@ Peers.prototype.acceptable = function (peers) {
...
+					broadhash: null, //modules.system.getBroadhash(),
 					height: 1
 				});
-
-				return setImmediate(eachCb);
+				setImmediate(eachCb);
```
Comment (karmacoma) [inline modules/peers.js]: Let's stick to returning explicitly.

**Comment by karmacoma** [comment]
Requested changes and reorganization to be addressed by new pull request.

File: modules/peers.js (hunk 3 above)
Comment (4miners) [inline modules/peers.js]: Yes, `pgp.helpers` is not available without initialization.

File: modules/peers.js (hunk 2 above)
Comment (4miners) [inline modules/peers.js]: Ok.

File: modules/peers.js (hunk 7 above)
Comment (4miners) [inline modules/peers.js]: That function is redesigned.

File: modules/peers.js (hunk 8 above)
Comment (4miners) [inline modules/peers.js]: Sure.

**Pull request closed without merging**